tf2_gnn/models/graph_regression_task.py

```diff
@@ -26,46 +34,121 @@ def __init__(self, params: Dict[str, Any], dataset: GraphDataset, name: str = No
         super().__init__(params, dataset=dataset, name=name)
         self._node_to_graph_aggregation = None
 
+        # Construct sublayers:
+        self._weighted_avg_of_nodes_to_graph_repr = WeightedSumGraphRepresentation(
+            graph_representation_size=self._params["graph_representation_size"] // 2,
+            num_heads=self._params["graph_aggregation_num_heads"] // 2,
+            weighting_fun="softmax",
+            scoring_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            scoring_mlp_dropout_rate=self._params["graph_aggregation_dropout_rate"],
+            scoring_mlp_activation_fun="elu",
+            transformation_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            transformation_mlp_dropout_rate=self._params[
+                "graph_aggregation_dropout_rate"
+            ],
+            transformation_mlp_activation_fun="elu",
+        )
+        self._weighted_sum_of_nodes_to_graph_repr = WeightedSumGraphRepresentation(
+            graph_representation_size=self._params["graph_representation_size"] // 2,
+            num_heads=self._params["graph_aggregation_num_heads"] // 2,
+            weighting_fun="sigmoid",
+            scoring_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            scoring_mlp_dropout_rate=self._params["graph_aggregation_dropout_rate"],
+            scoring_mlp_activation_fun="elu",
+            transformation_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            transformation_mlp_dropout_rate=self._params[
+                "graph_aggregation_dropout_rate"
+            ],
+            transformation_mlp_activation_fun="elu",
+        )
+
+        self._regression_mlp = MLP(
+            out_size=1,
+            hidden_layers=self._params["regression_mlp_layers"],
+            dropout_rate=self._params["regression_mlp_dropout"],
+            use_biases=True,
+            activation_fun=tf.nn.relu,
+        )
+
     def build(self, input_shapes):
+        # We get the initial GNN input + results for all layers:
+        node_repr_size = (
+            input_shapes["node_features"][-1]
+            + self._params["gnn_hidden_dim"] * self._params["gnn_num_layers"]
+        )
```

Comment:
What happens if `use_intermediate_gnn_results = False`? Shouldn't it then be just
```python
node_repr_size = input_shapes["node_features"][-1] + self._params["gnn_hidden_dim"]
```